### PR TITLE
create single product route

### PR DESCRIPTION
**Pull request by rawand1998**

### CODE REVIEW

File: server/controllers/getProduct.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+const getProduct = (req,res)=>{
+    const {product_id}=req.params;
```
Comment (umarsalah) [inline server/controllers/getProduct.js:3]: use camelCase for your variables naming ```productId``` not ```product_id```

File: server/controllers/getProduct.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    getSingleProduct(product_id)
+    .then(() => res.json({ status: 'scusses' }))
+    .catch((err) => res.json({ status: 'err', error: err }));
```
Comment (umarsalah) [inline server/controllers/getProduct.js:6]: here you get the data from the database then you send a message you don't send the data!

**Review by umarsalah — DISMISSED** [review]

File: server/controllers/getProduct.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+const getProduct = (req,res)=>{
+    const {productId}=req.params;
+    getSingleProduct(productId)
+    .then((data) => res.json(data.rows))
```
Comment (Mu7ammadAbed) [inline server/controllers/getProduct.js:5]: I think you need to send the first element of the rows array, check it and destructure it
```suggestion
    .then(({ rows }) => res.json(rows[0]))
```

File: server/database/queries/getSingleProduct.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+const connection = require('../connection');
+
+const getSingleProduct =(productId)=>connection.query('select * from product where product_id=$1',[productId])
```
Comment (Mu7ammadAbed) [inline server/database/queries/getSingleProduct.js:3]:
```suggestion
const getSingleProduct =(productId)=>connection.query('SELECT * FROM product WHERE product_id=$1', [productId])
```

File: server/routes/index.js [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 const addUserHandler = require('./addUser');
 const addCardHandler = require('./addCard');
-
+const {getProduct} = require('../controllers/index')
 const router = express.Router();
 
 router.post('/singup', addUserHandler);
 router.post('/addcard', addCardHandler);
-
+router.get('/api/product/:id',getProduct)
```
Comment (Mu7ammadAbed) [inline server/routes/index.js:9]: I believe you called it `productId`
```suggestion
router.get('/api/product/:productId',getProduct)
```

File: server/controllers/getProduct.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+
+const getProduct = (req,res)=>{
+    const {productId}=req.params;
```
Comment (Mu7ammadAbed) [inline server/controllers/getProduct.js:3]: Validate this to check it's a number, you can do a simple validation by just doing this
```js
if (productId > 0) {
...
} else {
res.status(400).json({err...})
}
```

**Review by NourAdawi — APPROVED** [review]

**Review by muhammadharoun — APPROVED** [review]

**Review by mohammedsalah7 — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**